### PR TITLE
TW-1923: Improve search screen

### PR DESCRIPTION
**Pull request by hieutbui**
## Ticket
- #1923 

## Root cause
- Currently, when search message is empty, search controller doesn't check for keyword is Matrix ID

## Solution
- When search message is empty, check if keyword is Matrix ID then hide empty widget.

## Test recommendations
- Search keywords that is a valid Matrix ID 
- Search keywords that is a valid Matrix ID but not complete (Ex: @test:domain.co)
- Send message with a Matrix ID inside a room then search for this Matrix ID

## Resolved
- Web:

https://github.com/user-attachments/assets/5a4374c7-98c8-4295-9d6a-0f6c324d6c61

- Android:
- IOS:

### CODE REVIEW

**Comment by sherlockvn** [comment]
Screenshot 2024-07-25 at 11 02 37

File: lib/pages/search/search_view.dart [hunk 1]
```diff
@@ -73,8 +73,10 @@ class SearchView extends StatelessWidget {
                 searchController.serverSearchController.searchResultsNotifier,
             builder: ((context, searchResults, child) {
               if (searchResults is PresentationServerSideEmptySearch) {
+                final keyword = searchController.textEditingController.text;
                 if (searchController.searchContactAndRecentChatController!
-                    .recentAndContactsNotifier.value.isNotEmpty) {
+                        .recentAndContactsNotifier.value.isNotEmpty ||
+                    (keyword.isValidMatrixId && keyword.startsWith('@'))) {
```
Comment (sherlockvn) [inline lib/pages/search/search_view.dart:79]: better to make a function

File: lib/pages/search/server_search_view.dart [hunk 2]
```diff
@@ -28,11 +28,13 @@ class ServerSearchMessagesList extends StatelessWidget {
             searchController.serverSearchController.searchResultsNotifier,
         builder: (context, serverSearchNotifier, child) {
           if (serverSearchNotifier is PresentationServerSideEmptySearch) {
+            final keyword = searchController.textEditingController.text;
             if (searchController.searchContactAndRecentChatController!
-                .recentAndContactsNotifier.value.isNotEmpty) {
-              return const SizedBox.shrink();
+                    .recentAndContactsNotifier.value.isEmpty &&
+                !(keyword.isValidMatrixId && keyword.startsWith('@'))) {
+              return child!;
```
Comment (sherlockvn) [inline lib/pages/search/server_search_view.dart:35]: same

**Comment by sherlockvn** [comment]
not in the DOD, but i think this case, we should remove the `Messages` section too
image

**Comment by KhaledNjim** [comment]
> not in the DOD, but i think this case, we should remove the `Messages` section too image

Updated

image

**Comment by nqhhdev** [comment]
I tested it on mobile, web works well

**Comment by nqhhdev** [comment]
I have one concern about padding in chat search on web. WDYT @hoangdat @KhaledNjim 

- Currently 

Screenshot 2024-11-25 at 10 23 57

- After updated

Screenshot 2024-11-25 at 10 22 32

**Comment by hoangdat** [comment]
> I have one concern about padding in chat search on web. WDYT @hoangdat @KhaledNjim

please base on the new style for this.

**Comment by nqhhdev** [comment]
Reverted style
Screenshot 2024-12-04 at 23 48 15

**Comment by KhaledNjim** [comment]
According to the notion page, the explanation text when no results are found should be removed, its still present in some screens
 - When searching for a contact while creating a new chat
 

https://github.com/user-attachments/assets/bea07524-85fb-494a-bf2b-4305521c1878


  - When searching in contacts screen
  
image

**Comment by nqhhdev** [comment]
Hi @hoangdat  @KhaledNjim, I updated the search result in contact screen, add members into group and create a  new chat.
Pls help me review and test it again. Thanks all

https://github.com/user-attachments/assets/80effcdf-1b92-4ce6-88b1-9c266a85d244

File: lib/pages/search/search.dart [hunk 3]
```diff
@@ -58,6 +58,9 @@ class SearchController extends State<Search> {
 
   String get searchWord => textEditingController.text;
 
+  bool get isSearchWordUserId =>
```
Comment (hoangdat) [inline lib/pages/search/search.dart:61]: Can you find a better name for this? Still not understand what is it? What is use for?

**Pull request merged**